### DISCUSSION TITLE
Init form_class

### DISCUSSION THREAD

**Discussion by moas** [graphql-python/graphene-django#1174] [Q&A] [answered]
Hi
How To set  init data for form_class in djangoformmutation ? I try to use django auth forms PasswordChangeForm and this form need 'user' instance in __init__. So djangoformmutation crash in __init_subclass_with_meta__ execution

**Comment by dopeboy** [comment]
Can you provide some starter code here to help us?

**Comment by zbyte64** [comment]
DjangoFormMuation has a method: `def get_form_kwargs(cls, root, info, **input)` ; this is what you want to override, have it return a dictionary with the desired keyword arguments to be passed to the form init.

**Comment by nikolaik** [comment, marked as answer]
You can do something like this: 
```python
from django.contrib.auth.forms import PasswordChangeForm
from graphene_django.forms.mutation import DjangoFormMutation


class MyPasswordChangeForm(PasswordChangeForm):
    def __init__(self, *args, **kwargs):
        user = kwargs.pop('user', None)
        super().__init__(user, *args, **kwargs)


class ChangePassword(DjangoFormMutation):
    class Meta:
        form_class = MyPasswordChangeForm

    @classmethod
    def get_form_kwargs(cls, root, info, **mutation_input):
        return {**super().get_form_kwargs(root, info, **mutation_input), 'user': info.context.user}
```
